stdlib/public/core/EmbeddedPrint.swift

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftShims
+
+// TODO: This is all a stop-gap so that at least some types are printable in
+// embedded Swift, in an embedded-programming friendly way (we mainly need
+// printing to not need to heap allocate).
+
+@_silgen_name("putchar")
+@discardableResult
+func putchar(_: CInt) -> CInt
+
+public func print(_ string: StaticString, terminator: StaticString = "\n") {
+  var p = string.utf8Start
+  while p.pointee != 0 {
+    putchar(CInt(p.pointee))
+    p += 1
+  }
+  p = terminator.utf8Start
+  while p.pointee != 0 {
+    putchar(CInt(p.pointee))
+    p += 1
+  }
+}
+
+func printCharacters(_ buf: UnsafeRawBufferPointer) {
+  for c in buf {
+    putchar(CInt(c))
+  }
+}
+
+func printCharacters(_ buf: UnsafeBufferPointer<UInt8>) {
+  printCharacters(UnsafeRawBufferPointer(buf))
+}
+
+extension BinaryInteger {
+  func writeToStdout(radix: Int = 10) {
```

Comment:
since the radix doesn't apply to common bases like base 16, perhaps we should just not have anything but base 10 and hard code the radix for the function to 10

Reply:
done, hardcoded